snapcraft/plugins/nodejs.py

```diff
@@ -142,7 +141,7 @@ def _npm_install(self, rootdir):
 
 
 def _get_nodejs_base(node_engine):
-    machine = platform.machine()
+    machine = _options._get_platform_architecture()
```

Comment:
hmm, can't you pass `project_options.deb_arch` and change the map to use `deb_arch` instead?